### PR TITLE
Add support for working with remote repositories.

### PR DESCRIPTION
**Pull request by adithyaov**
See #30

### CODE REVIEW

**Review by pranaysashank — APPROVED** [review]

File: packcheck.sh
Comment (adithyaov) [reply packcheck.sh:1293]: Git is only required if we use options related to git.

File: README.md
Comment (harendra-kumar) [inline README.md:344]: It is sneaky to create a hidden .packcheck.remote in the current directory. Especially because the command can be run from anywhere. We could be more explicit and clone the repo visibly in the current directory and then run packcheck inside that.

File: README.md
Comment (harendra-kumar) [inline README.md:312]: Maybe `GIT_SOURCE_REV` and `GIT_SOURCE_MERGE`?

File: packcheck.sh
Comment (harendra-kumar) [inline packcheck.sh:1648]: We may want to do some sanity checks e.g. GIT_REV/GIT_MERGE makes sense only if GIT_SOURCE is specified.

**Review by harendra-kumar — COMMENTED** [review]
Should we keep packcheck as purely a CI script and use a separate script for cloning and merging a git repository? We can call packcheck from that script after cloning and merging.

One reason for this is that packcheck uses a very crude environment variable based interface to work better with the CI systems. Whereas the git cloning and merging script could use a better CLI interface with regular `--git-source`, `--git-rev` type options instead of using env vars.

At this point it seems convenient to add just a few more env vars to packcheck itself, but I am not sure if this becomes more and more complex in future.

File: packcheck.sh
Comment (adithyaov) [reply packcheck.sh:1648]: If `GIT_SOURCE` isn't specified `GIT_REV` & `GIT_MERGE` are not used. Should we error out or print a warning?

**Comment by adithyaov** [comment]
> Should we keep packcheck as purely a CI script and use a separate script for cloning and merging a git repository? We can call packcheck from that script after cloning and merging.

Yeah, I was going to try to split `packcheck.sh` into reasonable files that can be used independently.

We can hold this off for now and get back to this once we figure out how to make packcheck scalable.

File: README.md
Comment (adithyaov) [reply README.md:344]: We can maybe add a `GIT_CLONE_TO`.

**Comment by harendra-kumar** [comment]
It may not be a good idea to break packcheck.sh into multiple scripts, it will be harder to deploy. You will have to download different parts and make them work together by including parts into an overall script.

However, it makes sense to keep the remote CI script separate. I would propose that we have a `packcheck-remote` script and push this functionality into that. If we put this in packcheck.sh temporarily then we cannot remove it. The purpose of `packcheck-remote` would be to just pull a remote repo, merge etc. and then call packcheck.sh on it. It would be modular.

**Comment by adithyaov** [comment]
Closing in favor of #36

**Pull request closed without merging**
